### PR TITLE
[PVR][EPG] Make EPG a subcomponent of PVR

### PR DESCRIPTION
**Pull request by ksooo**
As discussed here https://github.com/xbmc/xbmc/pull/11964#pullrequestreview-32785333 . This PR also eliminates the global static instance for the epg container.

This change has been runtime tested on macOS, latest kodi master.

@FernetMenta mind doing the code review.

### CODE REVIEW

File: xbmc/pvr/PVRTypes.h [hunk 1]
```diff
@@ -59,5 +59,14 @@ namespace PVR
   class CPVRGUIActions;
   typedef std::shared_ptr<CPVRGUIActions> CPVRGUIActionsPtr;
 
+  class CPVREpgContainer;
+  typedef std::shared_ptr<CPVREpgContainer> CPVREpgContainerPtr;
```
Comment (FernetMenta) [inline xbmc/pvr/PVRTypes.h]: why is this a shared pointer? imo epgContainer could just be a plain member of PVRManager and clients could ask for a reference. it has the same lifetime as the pvrManager object.

**Review by FernetMenta — APPROVED** [review]

File: xbmc/pvr/PVRTypes.h (hunk 1 above)
Comment (ksooo) [inline xbmc/pvr/PVRTypes.h]: This is a good point. I will change that.

**Comment by ksooo** [comment]
@FernetMenta incorporated your review comment.

jenkins build this please

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
jenkins error is unrelated.

**Comment by ksooo** [comment]
Heads-up: if nobody objects I will merge this tomorrow.

**Pull request merged**